lib/matplotlib/legend.py

```diff
@@ -824,55 +824,34 @@ def _init_legend_box(self, handles, labels, markerfirst=True):
 
     def _auto_legend_data(self):
         """
-        Returns list of vertices and extents covered by the plot.
+        Return display coordinates for hit testing for "best" positioning.
 
-        Returns a two long list.
-
-        First element is a list of (x, y) vertices (in
-        display-coordinates) covered by all the lines and line
-        collections, in the legend's handles.
-
-        Second element is a list of bounding boxes for all the patches in
-        the legend's handles.
+        Returns
+        -------
+        vertices
+            List of (x, y) vertices of all lines.
+        bboxes
+            List of bounding boxes of all patches.
+        lines
+            List of `.Path` corresponding to each line.
+        offsets
+            List of (x, y) offsets of all collection.
         """
-        # should always hold because function is only called internally
-        assert self.isaxes
-
+        assert self.isaxes  # always holds, as this is only called internally
         ax = self.parent
-        bboxes = []
-        lines = []
+        lines = [line.get_transform().transform_path(line.get_path())
+                 for line in ax.lines]
+        vertices = np.concatenate([l.vertices for l in lines]) if lines else []
```

Comment:
Before the empty case returned an empty array. Are we sure we can use an empty list instead?

Reply:
Yes, this is only passed later to `legendBox.count_contains` which casts its input to an array anyways.

Reply:
For consistency, I would be slightly more at peace if the return value was always a string. Since this is not performance critical, it should not matter. Bu I won't insist.

Reply:
In that case I may just as well not construct the vertices array at all (saving a concatenate, which *may* needlessly allocate a lot of memory) and just count_countains() on each line.vertices separately (you have to iterate once over all lines anyways, it shouldn't matter whether you do it when concatenating or when calling count_contains()).  Edited accordingly.